### PR TITLE
added limit to request

### LINKED ISSUE #42: Add parameter to polls/all route to fetch only x polls
On the startpage, we would like to have only 10 polls. At this point, we call all polls and filter on frontend the amount of displayed polls.

Add an optional query parameter `limit` to set the maximum amount of polls. If the client does not append these query parameter, all polls should be returned.

### PR DESCRIPTION
**Pull request by GrafBunsen**
an optional parameter named limit was added. when /api/polls/all is given the parameter it only retourns the amount of polls given by the parameter. closing #42

### CODE REVIEW

File: app/Http/Controllers/PollController.php [hunk 1]
```diff
@@ -56,13 +56,14 @@ public function getPoll(Request $request, int $id): JsonResponse
...
+
         return response()->json($convertedPoll, status: 200);
     }
 
-    public function getAll(): JsonResponse
+    public function getAll(Request $request): JsonResponse
     {
-        $polls = Poll::all();
+        $limit = $request->input("limit");
+        $polls = Poll::limit($limit)->get();
```
Comment (martincodes-de) [inline app/Http/Controllers/PollController.php]: please add a way to provide all polls. the solution is imo coincidental

**Review by martincodes-de — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/PollController.php (hunk 1 above)
Comment (GrafBunsen) [reply app/Http/Controllers/PollController.php]: done

File: app/Http/Controllers/PollController.php [hunk 2]
```diff
@@ -56,13 +55,18 @@ public function getPoll(Request $request, int $id): JsonResponse
...
+
         return response()->json($convertedPoll, status: 200);
     }
 
-    public function getAll(): JsonResponse
+    public function getAll(Request $request): JsonResponse
     {
-        $polls = Poll::all();
+        $limit = $request->input("limit");
+        if($limit== null){
```
Comment (martincodes-de) [inline app/Http/Controllers/PollController.php]: pls use ternary operator

**Review by martincodes-de — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/PollController.php (hunk 2 above)
Comment (GrafBunsen) [reply app/Http/Controllers/PollController.php]: done

File: app/Http/Controllers/PollController.php [hunk 3]
```diff
@@ -56,13 +55,14 @@ public function getPoll(Request $request, int $id): JsonResponse
...
+
         return response()->json($convertedPoll, status: 200);
     }
 
-    public function getAll(): JsonResponse
+    public function getAll(Request $request): JsonResponse
     {
-        $polls = Poll::all();
+        $limit = $request->input("limit");
+        $polls =$limit==null? Poll::all() : Poll::limit($limit)->get();
```
Comment (martincodes-de) [inline app/Http/Controllers/PollController.php]:
```suggestion
        $polls = is_null($limit) ? Poll::all() : Poll::limit($limit)->get();
```

**Review by martincodes-de — CHANGES_REQUESTED** [review]

**Review by martincodes-de — APPROVED** [review]

**Review by emso-sp — APPROVED** [review]

**Pull request merged**